test/Main.hs

```diff
@@ -73,6 +82,58 @@ spec = do
         dpiTimeStampToUTCDPITimeStamp dpi `shouldBe` expected
 
       it "Should roundtrip UTCTime through DPITimestamp (w/ nanos -- not picos) " $ \_ -> do
-        property $ \tod day (nanos :: Nano) -> do
-          let utc = UTCTime day $ timeOfDayToTime tod { todSec = realToFrac nanos }
+        property $ \(UTCTimeNanos utc) -> do
           utc `shouldBe` dpiTimeStampToUTCTime (utcTimeToDPITimestamp utc)
+
+    describe "Roundtrip tests" $ do
+      it "Should round trip random values from a table" $ \conn -> do
+        property $ \x@TestTable{..} -> do
+          execute_ conn "create table test (a varchar(300), b number (12,0), c number (12,0), d number (12,0) null, e timestamp, f number (38,28))"
+          execute conn "insert into test values (:1,:2,:3,:4,:5,:6)" x
+          y  <- query_ conn "select * from test"
+          execute_ conn "drop table test"
+          [x] `shouldBe` y
```

Comment:
Also the arguments to `shouldBe` shouldBe flipped since it's `shouldBe actual expected`, the failure message we get right now mixes up what was expected and what we got, which made tracking the issue harder.

Reply:
fixed